### PR TITLE
Added new block definitions, whitespace control and expression caching

### PR DESCRIPTION
**Pull request by fschwaiger**
Using this in my own tools, I felt this was lacking the following features, which I implement in this PR:

1. Whitespace control
2. New block types: `elseif`, `switch`, `case`, `let`, `assert`, `error`, `include`, `otherwise`
3. Expression cache for much faster rendering speeds.
4. File content cache in `LFileTemplate`.

Two functions have been renamed / deleted. I can modify their names if you prefer to exclude that from the diff.

Also added myself to the license file, if that is the correct way to do that.

### CODE REVIEW

**Comment by comkieffer** [comment]
Hi, Thanks for the PR. 

I'd entirely forgotten about this project so I'm glad that it was helpful to you. i'm very supportive of your changes. Before I merge them, do you have an example with the new features that I can use to test ? 

Thanks

**Comment by fschwaiger** [comment]
It is not a full test example, but I can provide the following:

- there are new snippets in the README
- there are new snippets in the test folder
- there are some files in https://gitlab.com/tum-fsd/tico/-/tree/main/resources/template, e.g. https://gitlab.com/tum-fsd/tico/-/blob/main/resources/template/gitlab-ci.template is a good one

I also noticed that since I am using property validation syntax, the MATLAB base version probably needs to be raised. This would be a major version bump.

File: src/README.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
+# The Lobster Template Engine
```
Comment (comkieffer) [inline src/README.md:1]: The README seems to be duplicated.

**Comment by comkieffer** [comment]
Other than the README being duplicated, this looks great. I'll be away for the next couple of weeks but I'll merge it when I come back. 

Thanks for the work you've done.

**Pull request merged**